lib/cli/resolve-configuration-path.js

```diff
@@ -0,0 +1,71 @@
+'use strict';
+
+const path = require('path');
+const fs = require('fs').promises;
+const ServerlessError = require('../serverless-error');
+const fileExists = require('../utils/fs/fileExists');
+
+const supportedExtensions = new Set(['yml', 'yaml', 'json', 'js', 'ts']);
+
+module.exports = async () => {
+  const customConfigName = (() => {
+    const configParamIndex = (() => {
+      const paramIndex = process.argv.indexOf('--config');
+      if (paramIndex !== -1) return paramIndex;
+      return process.argv.indexOf('-c');
+    })();
+    return configParamIndex === -1 ? null : process.argv[configParamIndex + 1];
+  })();
+
+  if (customConfigName) {
+    if (path.isAbsolute(customConfigName)) {
```

Comment:
Is there any reason why a configuration path cannot be absolute? You already handle that nicely by using `path.resolve` just below (compared to the previous `path.join`)?

After all, this just makes it ever so slightly harder: Now I need to take my absolute path, and relativize it against the current directory, and then provide that ...

Reply:
@ankon Thanks for that comment, that's a very valid point. I've prepared a fix at: https://github.com/serverless/serverless/pull/8793